Docker.DotNet/DockerPipeStream.cs

```diff
@@ -0,0 +1,124 @@
+using System;
+using System.IO;
+using System.IO.Pipes;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Net.Http.Client;
+using Microsoft.Win32.SafeHandles;
+
+namespace Docker.DotNet
+{
+    internal class DockerPipeStream : WriteClosableStream
+    {
+        private PipeStream _stream;
+        private EventWaitHandle _event = new EventWaitHandle(false, EventResetMode.AutoReset);
+
+        public DockerPipeStream(PipeStream stream)
+        {
+            _stream = stream;
+        }
+
+        public override bool CanCloseWrite => true;
+
+        public override bool CanRead => true;
+
+        public override bool CanSeek => false;
+
+        public override bool CanWrite => true;
+
+        public override long Length
```

Comment:
This doesn't work since throw foo is a statement not an expression. It turns out the brace style fat arrow is not supported in this context, either.